src/main/scala/org/broadinstitute/dsde/firecloud/core/ProfileClient.scala

```diff
@@ -208,7 +209,10 @@ class ProfileClientActor(requestContext: RequestContext) extends Actor with Fire
           RawlsGroupMemberList(userSubjectIds = Some(subjectIds))
         }}
       }
-      case Some(id) => Future.successful(RawlsGroupMemberList(userSubjectIds = Some(Seq(id))))
+      case Some((fcUser, nihUser)) =>
+        if(whitelist contains nihUser) Future.successful(RawlsGroupMemberList(userSubjectIds = Some(Seq(fcUser))))
+        else Future.failed(throw new FireCloudExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden,
```

Comment:
that technically wouldn't grant them access to the TCGA data because it wouldn't put them in the dbGap group (but it would still call UPDATE on the group, with the empty payload)

the issue is that in `UpdateNIHLinkAndSyncSelf`, we can't really tell the difference between when we've updated the group with an empty `RawlsGroupMemberList` and when we've updated the group with a populated `RawlsGroupMemberList`

as a result, the thurloe entries `lastLinkTime`, `linkExpireTime`, and `linkedNihUsername` would all still get updated- and this would trigger the UI to say "You have successfully linked, your access will be granted shortly"

so my choice (mainly in the interest of time) was to go nuclear and throw an exception instead of rewriting UpdateNIHLinkAndSyncSelf. the exception allows us to fail quickly and not store any inaccurate junk in thurloe